### PR TITLE
[TD]add extensions for drag and drop

### PR DESCRIPTION
**Pull request by WandererFan**
TechDraw did not always handle drag and drop correctly.  This PR prevents pages from  being broken when Templates or Views are dragged and dropped.

Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [ ]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8097">pipeline status</a> for feature branch [PR_8097](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8097). Pipeline [733038380](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/733038380) was triggered at [71ed0e0](https://github.com/FreeCAD/FreeCAD/pull/8097/commits/71ed0e05831506a73fec4aa613d03136bdf65054). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/TechDraw/App/DrawPage.cpp [hunk 1]
```diff
@@ -470,6 +470,19 @@ bool DrawPage::canUpdate() const
     return result;
 }
 
+//true if object belongs to this page
+bool DrawPage::hasObject(App::DocumentObject* obj)
+{
+    for (auto& outObj : getOutList()) {
+        //TODO: check if pointer comparision is reliable enough
```
Comment (luzpaz) [inline src/Mod/TechDraw/App/DrawPage.cpp:477]: comparision ==> comparison

**Comment by WandererFan** [comment]
Are you on Windows?  I just built master branch on Linux without problems.

**Comment by WandererFan** [comment]
Not sure who my last reply went to...
Are you on Windows?  I just built master branch without problems.

**Comment by WandererFan** [comment]
I think I know what it is.  I'll try a fix on a Windows VM later today.

**Comment by WandererFan** [comment]
Should be fixed by PR #8121 / commit: d1cf5f9c88

**Pull request merged**